src/utils/transaction.ts

```diff
@@ -204,8 +203,20 @@ const transaction = {
       });
     }
 
-    if (tx.version === NANO_CONTRACTS_VERSION || tx.version === ON_CHAIN_BLUEPRINTS_VERSION) {
-      const { pubkey } = tx as NanoContract | OnChainBlueprint;
+    let pubkey: Buffer | null = null;
+
+    if (tx.isNanoContract()) {
```

Comment:
The old implementation would check that the transaction is a nano contract and extract the pubkey for the extra signature.
The current implementation has 2 ways to extract this pubkey, maybe we should have a separate method 
`getNcCallerPubkey` with this logic since it may be useful in other places.

Reply:
It's done in another commit with similar changes: https://github.com/HathorNetwork/hathor-wallet-lib/pull/842/commits/1e0f72f780560ece48ab99251f159585e16674f8